research-info.lua

```diff
@@ -0,0 +1,69 @@
+local researchInfo = reqscript("internal/research-info/info")
+
+local function sortResearchDataByPercAndPondered(data_list)
+    table.sort(data_list, function(a, b)
+        if tonumber(a.research_percentage) == tonumber(b.research_percentage) then
+            return a.times_pondered > b.times_pondered
+        end
+        return tonumber(a.research_percentage) > tonumber(b.research_percentage)
+    end)
+
+    return data_list
+end
+
+local function printResearchData(data)
+    if not data then
+        print("No data to display.")
+        return
+    end
+
+    if data.research_points == 0 and data.knowledge_goal == "" and data.times_pondered == 0 then
+        return
+    end
+
+    print("==================")
+    print("Dwarf: " .. data.name)
+    print("\tKnowledge Goal Category: " .. data.knowledge_goal_category)
+    print("\tKnowledge Goal: " .. data.knowledge_goal)
+    print("\tResearch Points: " .. data.research_points .. "/100000")
+    print("\tResearch Percentage: " .. data.research_percentage .. "%")
+    print("\tTimes Pondered: " .. data.times_pondered)
+
+    if #data.research_topics > 0 then
+        print("Knowledge:")
+        for _, topic in ipairs(data.research_topics) do
+            print("\t" .. topic.name .. ": " .. topic.info)
+        end
+    end
+end
+
+--
+local unit  = dfhack.units.getCitizens()
+local selected_unit = dfhack.gui.getSelectedUnit()
+
+if unit then
```

Comment:
`dfhack.units.getCitizens` returns a table, and even the empty table (which this will never return in an active fort) is a "truthy" value in lua. So this condition can never evaluate to false.